operator/controllers/resources/role_controller.go

```diff
@@ -82,7 +91,8 @@ func (r *RoleReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	// To handle this more gracefully we unmarshall first in an unstructured object.
 	// The unstructured object will be converted later to a typed one, in r.UpsertExternal.
 	// See `/operator/crdgen/schemagen.go` and https://github.com/gravitational/teleport/issues/15204 for context
-	obj := getUnstructuredObjectFromGVK(teleportRoleGVK)
+	// TODO: (Check how to handle multiple versions)
+	obj := getUnstructuredObjectFromGVK(teleportRoleGVKV5)
```

Comment:
If I understand correctly, the controller only listens for v5 objects. I'm unsure if listening to `v5` is enough to get notified when someone edits a role through the `v6` API. I'll have to run more tests to understand how the informer behaves with multiple versions of the same resource.